lnwallet/btcwallet/btcwallet.go

```diff
@@ -542,7 +557,20 @@ func (b *BtcWallet) ListTransactionDetails() ([]*lnwallet.TransactionDetail, err
 		txDetails = append(txDetails, detail)
 	}
 
-	return txDetails, nil
+	// Return empty transaction list, if offset is more than all transactions.
```

Comment:
Defaults should _also_ be enforced at the RPC server level. This way, even older RPC clients will be able to properly take advantage of the new code w/o updating their node.